src/main/blackbox/blackbox.c

```diff
@@ -914,6 +914,39 @@ static bool sendFieldDefinition(const char * const *headerNames, unsigned int he
 }
 
 /**
+ * Encode and transmit PID
+ */
+static uint32_t blackboxWritePID(uint8_t pid)
+{
+    uint8_t kP, kI, kD;
+
+    if (IS_PID_CONTROLLER_FP_BASED(currentProfile->pidProfile.pidController)) {
+        if (pid < 3) {
+            kP = constrain(lrintf(currentProfile->pidProfile.P_f[pid] * 10.0f), 0, 250);
```

Comment:
lrintf returns integer, constrain() takes integer input, I don't see this as a problem. 
I re-used the code from MSP: https://github.com/cleanflight/cleanflight/blob/master/src/main/io/serial_msp.c#L911
Theproblem is that we have Luxfloat, that does not use P8,I8,D8, but uses it's own parameters for rate PIDs and PIDLEVEL. This code takes care of that.